src/main/java/ch/njol/skript/conditions/CondLeashHolder.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.conditions;
+
+import java.util.Map;
+
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.EnchantmentType;
+import ch.njol.util.Checker;
+import ch.njol.util.Kleenean;
+
+@Name("Has Leash Holder")
+@Description("Checks whether entities have leash holders.")
+@Examples("target entity doesn't have a leash holder")
+@Since("INSERT VERSION")
+public class CondLeashHolder extends Condition {
+	
+	static {
+		Skript.registerCondition(CondLeashHolder.class,
+				"%livingentities% (has|do[es] [have]|contains) [a] leash holder [from %-entity%]",
+				"%livingentities% (has([n't]| not [got])|do[es](n't| not) (contain|have)) [a] leash holder [from %-entity%]");
```

Comment:
![image](https://user-images.githubusercontent.com/12614053/44617541-6933d280-a864-11e8-9c71-c7c01b69d026.png)

Reply:
`event-entity contains leash holder from player` what?

Reply:
What are these patterns lol they are really bad. 
**`(do[es]n't|don't|do[es] not)`** is the currently used pattern and better imo.
And **`ha(s|ve)`**.

contains is also so absurd.

Reply:
Also make the class name CondHasLeash like others..

Reply:
These are worse